src/internal_events/datadog_agent.rs

```diff
@@ -0,0 +1,36 @@
+use crate::source_sender::ClosedError;
+use metrics::counter;
+use vector_core::internal_event::InternalEvent;
+
+#[derive(Debug)]
+pub struct DatadogAgentStreamError {
+    pub error: ClosedError,
+    pub count: usize,
+}
+
+impl InternalEvent for DatadogAgentStreamError {
+    fn emit_logs(&self) {
+        error!(
+            message = "Failed to forward events, downstream is closed.",
+            error = %self.error,
+            error_type = "stream",
```

Comment:
I recommend something like `stream_closed` for these, although that may be too confusing, as it sounds like maybe _upstream_ was closed, which is not what this is describing.